internal/gengapic/paging.go

```diff
@@ -49,6 +49,15 @@ func (g *generator) iterTypeOf(elemField *descriptor.FieldDescriptorProto) (*ite
 			return &iterType{}, err
 		}
 
+		eMsg, ok := eType.(*descriptor.DescriptorProto)
+		if !ok {
+			return nil, errors.E(nil, "cannot find message type %q, malformed descriptor", eType)
+		}
+		// TODO: handle map entries correctly
+		if eMsg.GetOptions().GetMapEntry() {
+			return nil, errors.E(nil, "iterating over maps not supported yet: %q", eType)
```

Comment:
Just to make sure, we won't really ever get to this code path today with `compute` because `getPagingFields` would say "this is not a paginated rpc", right? I'm just concerned with this breaking `compute` generation today.

Reply:
I gather from the tests  that a map field is represented as a repeated set of (key,value) tuples, so in that case the old version of `pagingFields()` would have recognized this as a paging field (assuming `page_size`, `page_token`, and `next_page_token` were there) and this iterator would have done something? Am I interpreting correctly?

That said, we're in preview, so we do have the ability to make breaking changes right now.

Reply:
I'm not concerned with breaking changes to the client surface, I'm concerned with failure during generation, breaking the build.

Reply:
Oh, OK.

@software-dov it would still be useful to have a companion PR with the generated code; it's easier top spot some issues there. And to Noah's point, were you able to run CI test locally to verify no breakages should occur?

Reply:
We don't have a `map<string, Message>` in Showcase at the moment (you were working on that), so this specific case wouldn't be exercised I believe. A chicken in the egg problem. I would say instead of returning a generator error here, add code to `getPagingFields` that skips those RPCs that only have a `map` for pagination, treating them as normal Unary RPCs. This will of course cause a breaking change when we implement mapped pagination if we were to release this skipping behavior. Not sure this is a big deal though given the Compute client is in alpha for us.

Reply:
[This is the hacky example](https://github.com/googleapis/gapic-generator-go/pull/716) of code that would skip mapped pagination.

Reply:
True, yes. Probably the best thing is to skip mapped pagination for right now (that will unblock my upcoming Showcase changes) and then implement it shortly after that.

My very slight inclination is to not do a new release until we have both max_results and mapped_pagination in, but I'm fine doing a release without mapped pagination and then issuing a breaking change with it. After all, we are still alpha and have forewarned users.

Reply:
> not do a new release until we have both max_results and mapped_pagination in

sgtm! But one reason we'd want to is that Showcase regen CI uses the latest released version of the generator. So without a released version that handles/skips mapped pagination, your changes will still fail. That's the chicken and the egg problem here.

Reply:
Fine by me. I'll add an escape hatch for mapped pagination and tweak the tests real quick.